### DISCUSSION TITLE
Default value for required radio buttons group using zod enum

### DISCUSSION THREAD

**Discussion by jrozbicki** [react-hook-form/react-hook-form#10313] [Q&A]
I'm trying to create form with radio buttons and I can't find good solution typescript-wise. There are few requirements:
- async data as default value
- zod for schema

I want it to not be selected at first, yet required for user. Since `option` is a zod enum, I can't initialize it with `""` or `null` or `undefined`.

Questions: 
Do you have any suggestions how to deal with that? 
Can't `values` from `useForm` be partial of FormValues? Does it have to require whole schema?

Created codesandbox with reproduction: https://codesandbox.io/s/blissful-resonance-g4o2gp?file=/src/App.tsx

Current solution in codesandbox works in runtime, but TS fails because `values` is not `FormValues`.

**Comment by kiocos** [comment]
I think this may be what you want:
https://codesandbox.io/s/priceless-jepsen-43cdk1?file=/src/App.tsx

Suggestions:
- to achieve "async data as default value", use [reset](https://react-hook-form.com/api/useform/reset/). I've done it in a useEffect, but you can do it next to the data returned in your application. When you use reset and pass data do it, the data passed will be "the new" defaultValues.
- instead of `values`, use `defaultValues` to initialize your defaultValues on your useForm.

**Reply by jrozbicki** [reply]
Thanks for the reply @kiocosta.

- Yeah, I didn't really like syntax of useeffect + reset, reset alongside data could be better, but if I had multiple sources I'd have to fallback to useeffect again.
- in order to use defaultValues I'd have render form only after I get the data, I don't want users to have jumpy experience (I know that could be mitigated, but I just want to display form).

Question remains: Does `values` have to take all fields at once? What's the reason behind it? Can't `values` just take `Partial<FormValues>`?

**Reply by jrozbicki** [reply]
Actually we've missed other important point. If we HAVE TO set `option` (because `values` requires all fields) and we still want to preserve "not selected" state initially, what we can set `option` to? 

`null`, `undefined` or `""` will cause ts error.
